urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/SaganBijection.java

```diff
@@ -0,0 +1,173 @@
+package permlib.examples;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import permlib.Permutation;
+import permlib.Permutations;
+import permlib.classes.PermutationClass;
+import permlib.utilities.BTree;
+
+/**
+ * Can we find a bijection between avoiders of 132[I_k, 1, D_m] and  of 
+ * 231[I_k,1,D_m] that preserves maj?
+ * 
+ * First guess is that we can fix the position of the maximum.
+ * 
+ * @author Michael Albert
+ */
+public class SaganBijection {
+
+    public static void main(String[] args) {
+        checkDescendingTrees();
+        // checkDesDistribution();
+    }
+    
+    private static void checkDescendingTrees() {
+        int k = 1;
+        int m = 1;
+        int maxN = 12;
+        int[] p = new int[k+m+1];
+        int[] q = new int[k+m+1];
+        for(int i = 0; i < k; i++) {
+            p[i] = i;
+            q[i] = m + i;
+        }
+        p[k] = k+m;
+        q[k] = k+m;
+        for(int i = k+1; i < k+m+1; i++) {
+            p[i] = k+m+k-i;
+            q[i] = m+k-i;
+        }
+        PermutationClass c132 = new PermutationClass(new Permutation(p));
+        PermutationClass c231 = new PermutationClass(new Permutation(q));
+        for(int n = k+m+2; n < maxN; n++) {
+            HashMap<BTree, Integer> t132 = new HashMap<>();
+            HashMap<BTree, Integer> t231 = new HashMap<>();
+            for(Permutation s : new Permutations(c132, n)) {
+                BTree ts = BTree.descendingTree(s);
+                if (!t132.containsKey(ts)) {
+                    t132.put(ts, 0);
+                }
+                t132.put(ts, t132.get(ts)+1);
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 47:53, 54:60.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.